util-core/src/main/scala/com/twitter/util/Future.scala

```diff
@@ -514,8 +518,8 @@ def join[%s](%s): Future[(%s)] = join(Seq(%s)) map { _ => (%s) }""".format(
     f: Seq[In] => Future[Seq[Out]]
   )(
     implicit timer: Timer
-  ): In => Future[Out] = {
-    new BatchExecutor[In, Out](sizeThreshold, timeThreshold, sizePercentile, f)
+  ): Batcher[In, Out] = {
```

Comment:
We will run some test internally to see if it breaks anything.
If it doesn't, I think it's fine to take the binary incompatible change, just update the CHANGES file and explain the change.
This class is not widely used enough to require to wait for a major release (we decided not to strictly following semantic versioning).